ibis/expr/operations/generic.py

```diff
@@ -221,6 +222,36 @@ class Least(CoalesceLike):
     pass
 
 
+_normalize = Dispatcher("_normalize")
+
+
+@_normalize.register(dt.DataType, object)
+def default(type: dt.DataType, value: object) -> object:
+    return value
+
+
+@_normalize.register(dt.Floating, (int, float))
+def _float(type: dt.Floating, value: float) -> float:
+    return float(value)
+
+
+@_normalize.register(dt.UUID, str)
+def _str_to_uuid(type: dt.UUID, value: str) -> uuid.UUID:
+    return uuid.UUID(value)
+
+
+@_normalize.register(dt.String, uuid.UUID)
+def _uuid_to_str(type: dt.String, value: uuid.UUID) -> str:
+    return str(value)
+
+
+@_normalize.register(dt.UUID, uuid.UUID)
+def _uuid_to_uuid(type: dt.UUID, value: uuid.UUID) -> uuid.UUID:
+    """Need this to override _uuid_to_str since dt.UUID is a child of
+    dt.String"""
```

Comment:
Yikes. I don't think that's correct, I'll make an issue.

Reply:
I can fix that here or open a separate PR